src/fastfield/reader.rs

```diff
@@ -158,27 +201,35 @@ impl<Item: FastValue, C: FastFieldCodecReader> FastFieldReaderCodecWrapper<Item,
     /// Opens a fast field given a file.
     pub fn open(file: FileSlice) -> crate::Result<Self> {
         let mut bytes = file.read_bytes()?;
-        let id = u8::deserialize(&mut bytes)?;
+        let header = read_header(&mut bytes);
+        let id = header.codec_id;
         assert_eq!(
             BitpackedFastFieldSerializer::ID,
             id,
             "Tried to open fast field as bitpacked encoded (id=1), but got serializer with \
              different id"
         );
-        Self::open_from_bytes(bytes)
+        Self::open_from_bytes(bytes, header.gcd, header.min_value)
     }
     /// Opens a fast field given the bytes.
-    pub fn open_from_bytes(bytes: OwnedBytes) -> crate::Result<Self> {
+    pub fn open_from_bytes(bytes: OwnedBytes, gcd: u64, min_value: u64) -> crate::Result<Self> {
         let reader = C::open_from_bytes(bytes.as_slice())?;
         Ok(FastFieldReaderCodecWrapper {
+            gcd,
+            min_value,
             reader,
             bytes,
             _phantom: PhantomData,
         })
     }
     #[inline]
     pub(crate) fn get_u64(&self, doc: u64) -> Item {
-        Item::from_u64(self.reader.get_u64(doc, self.bytes.as_slice()))
+        let mut data = self.reader.get_u64(doc, self.bytes.as_slice());
+        if self.gcd != 1 {
+            data *= self.gcd;
```

Comment:
yeah so GCD<Codec> as a codec remove this if statement, and avoid making GCD too special.